### ISSUE TITLE
Something strange happening with script/setup

### ISSUE THREAD

**Issue by electrobrain1982** [bachya/simplisafe-python#844]
Something strange is happening with multidict when the script executes with "script/setup".  

I get this error:

  - Updating multidict (5.2.0 -> 6.0.4): Failed

  ChefBuildError

  Backend subprocess exited when trying to invoke build_wheel

**To Reproduce**
I'm on python 3.12.4 on macOS, following the steps carefully here:

https://simplisafe-python.readthedocs.io/en/latest/usage.html#authentication

Any ideas?

**Comment by bachya** [comment]
Thanks for your patience. I'm unfortunately unable to reproduce this—from the latest `dev` branch:

```
simplisafe-python on  dev is 📦 v2024.01.0 via 🐍 v3.12.1 (.venv)
❯ script/setup
/opt/homebrew/bin/mise
mise all runtimes are installed
Requirement already satisfied: poetry in ./.venv/lib/python3.12/site-packages (1.7.1)
Requirement already satisfied: build<2.0.0,>=1.0.3 in ./.venv/lib/python3.12/site-packages (from poetry) (1.0.3)
Requirement already satisfied: cachecontrol<0.14.0,>=0.13.0 in ./.venv/lib/python3.12/site-packages (from cachecontrol[filecache]<0.14.0,>=0.13.0->poetry) (0.13.1)
Requirement already satisfied: cleo<3.0.0,>=2.1.0 in ./.venv/lib/python3.12/site-packages (from poetry) (2.1.0)
Requirement already satisfied: crashtest<0.5.0,>=0.4.1 in ./.venv/lib/python3.12/site-packages (from poetry) (0.4.1)
Requirement already satisfied: dulwich<0.22.0,>=0.21.2 in ./.venv/lib/python3.12/site-packages (from poetry) (0.21.7)
Requirement already satisfied: fastjsonschema<3.0.0,>=2.18.0 in ./.venv/lib/python3.12/site-packages (from poetry) (2.19.0)
Requirement already satisfied: installer<0.8.0,>=0.7.0 in ./.venv/lib/python3.12/site-packages (from poetry) (0.7.0)
Requirement already satisfied: keyring<25.0.0,>=24.0.0 in ./.venv/lib/python3.12/site-packages (from poetry) (24.3.0)
Requirement already satisfied: packaging>=20.5 in ./.venv/lib/python3.12/site-packages (from poetry) (23.0)
Requirement already satisfied: pexpect<5.0.0,>=4.7.0 in ./.venv/lib/python3.12/site-packages (from poetry) (4.9.0)
Requirement already satisfied: pkginfo<2.0.0,>=1.9.4 in ./.venv/lib/python3.12/site-packages (from poetry) (1.9.6)
Requirement already satisfied: platformdirs<4.0.0,>=3.0.0 in ./.venv/lib/python3.12/site-packages (from poetry) (3.10.0)
Requirement already satisfied: poetry-core==1.8.1 in ./.venv/lib/python3.12/site-packages (from poetry) (1.8.1)
Requirement already satisfied: poetry-plugin-export<2.0.0,>=1.6.0 in ./.venv/lib/python3.12/site-packages (from poetry) (1.6.0)
Requirement already satisfied: pyproject-hooks<2.0.0,>=1.0.0 in ./.venv/lib/python3.12/site-packages (from poetry) (1.0.0)
Requirement already satisfied: requests<3.0,>=2.26 in ./.venv/lib/python3.12/site-packages (from poetry) (2.32.3)
Requirement already satisfied: requests-toolbelt<2,>=0.9.1 in ./.venv/lib/python3.12/site-packages (from poetry) (1.0.0)
Requirement already satisfied: shellingham<2.0,>=1.5 in ./.venv/lib/python3.12/site-packages (from poetry) (1.5.4)
Requirement already satisfied: tomlkit<1.0.0,>=0.11.4 in ./.venv/lib/python3.12/site-packages (from poetry) (0.12.1)
Requirement already satisfied: trove-classifiers>=2022.5.19 in ./.venv/lib/python3.12/site-packages (from poetry) (2023.11.29)
Requirement already satisfied: virtualenv<21.0.0,>=20.23.0 in ./.venv/lib/python3.12/site-packages (from poetry) (20.24.3)
Requirement already satisfied: xattr<0.11.0,>=0.10.0 in ./.venv/lib/python3.12/site-packages (from poetry) (0.10.1)
Requirement already satisfied: msgpack>=0.5.2 in ./.venv/lib/python3.12/site-packages (from cachecontrol<0.14.0,>=0.13.0->cachecontrol[filecache]<0.14.0,>=0.13.0->poetry) (1.0.7)
Requirement already satisfied: filelock>=3.8.0 in ./.venv/lib/python3.12/site-packages (from cachecontrol[filecache]<0.14.0,>=0.13.0->poetry) (3.12.2)
Requirement already satisfied: rapidfuzz<4.0.0,>=3.0.0 in ./.venv/lib/python3.12/site-packages (from cleo<3.0.0,>=2.1.0->poetry) (3.5.2)
Requirement already satisfied: urllib3>=1.25 in ./.venv/lib/python3.12/site-packages (from dulwich<0.22.0,>=0.21.2->poetry) (2.2.2)
Requirement already satisfied: jaraco.classes in ./.venv/lib/python3.12/site-packages (from keyring<25.0.0,>=24.0.0->poetry) (3.3.0)
Requirement already satisfied: ptyprocess>=0.5 in ./.venv/lib/python3.12/site-packages (from pexpect<5.0.0,>=4.7.0->poetry) (0.7.0)
Requirement already satisfied: charset-normalizer<4,>=2 in ./.venv/lib/python3.12/site-packages (from requests<3.0,>=2.26->poetry) (3.2.0)
Requirement already satisfied: idna<4,>=2.5 in ./.venv/lib/python3.12/site-packages (from requests<3.0,>=2.26->poetry) (3.7)
Requirement already satisfied: certifi>=2017.4.17 in ./.venv/lib/python3.12/site-packages (from requests<3.0,>=2.26->poetry) (2024.7.4)
Requirement already satisfied: distlib<1,>=0.3.7 in ./.venv/lib/python3.12/site-packages (from virtualenv<21.0.0,>=20.23.0->poetry) (0.3.7)
Requirement already satisfied: cffi>=1.0 in ./.venv/lib/python3.12/site-packages (from xattr<0.11.0,>=0.10.0->poetry) (1.16.0)
Requirement already satisfied: pycparser in ./.venv/lib/python3.12/site-packages (from cffi>=1.0->xattr<0.11.0,>=0.10.0->poetry) (2.21)
Requirement already satisfied: more-itertools in ./.venv/lib/python3.12/site-packages (from jaraco.classes->keyring<25.0.0,>=24.0.0->poetry) (10.1.0)

[notice] A new release of pip is available: 24.1.2 -> 24.2
[notice] To update, run: pip install --upgrade pip
Installing dependencies from lock file

Package operations: 1 install, 14 updates, 0 removals

  • Updating babel (2.12.1 -> 2.15.0)
  • Updating docutils (0.18.1 -> 0.20.1)
  • Installing aiohappyeyeballs (2.3.2)
  • Updating pytest (8.2.2 -> 8.3.2)
  • Updating setuptools (68.1.2 -> 70.0.0)
  • Updating sphinx (7.3.7 -> 7.4.7)
  • Updating aiohttp (3.9.5 -> 3.10.0)
  • Updating astroid (3.2.2 -> 3.2.4)
  • Updating coverage (7.5.4 -> 7.6.0)
  • Updating pytest-asyncio (0.23.7 -> 0.23.8)
  • Updating mypy (1.10.1 -> 1.11.1)
  • Updating pre-commit (3.7.1 -> 3.8.0)
  • Updating pylint (3.2.5 -> 3.2.6)
  • Updating pyupgrade (3.16.0 -> 3.17.0)
  • Updating ruff (0.5.1 -> 0.5.5)

Installing the current project: simplisafe-python (2024.01.0)
pre-commit installed at .git/hooks/pre-commit
```

Are you still experiencing this issue?

**Comment by zombiebrak** [comment]
I was able to replicate this on a Proxmox VM running HAOS 2024.7.4.
```
- Downgrading multidict (6.0.5 -> 6.0.4): Failed
```
I was able to manually downgrade to 6.0.4.
```
pip install --force-reinstall -v "multidict==6.0.4"
```
I also had to manually downgrade frozenlist to 1.4.1.

After those downgrades, the setup script ran fine.
